scripts/deleteInactiveIntercomUsers.js

```diff
@@ -0,0 +1,65 @@
+// Delete Intercom users last seen over 30 days ago with no conversations
+
+'use strict';
+if (process.argv.length !== 3) {
+  console.log("Usage: node <script> <Intercom 'App ID:API write key'> ");
+  process.exit();
+}
+
+const intercomAppId = process.argv[2].split(':')[0];
+const intercomApiKey = process.argv[2].split(':')[1];
+
+const Promise = require('bluebird');
+const Intercom = require('intercom-client');
+const client = new Intercom.Client({ appId: intercomAppId, appApiKey: intercomApiKey });
+
+// NOTE: max 500 requests/minute, scroll closes after 1 minute and returns 100 user batches
+// So, scroll every 20s
+const scrollIntervalMilliseconds = 20 * 1000;
+
+const latestUpdate = new Date();
+latestUpdate.setUTCDate(latestUpdate.getUTCDate() - 30);
+
+client.users.scroll.each({}, function (response) {
+  console.log(`${new Date().toISOString()} DEBUG: scrolling ${scrollIntervalMilliseconds / 1000} seconds..`);
+  return new Promise((resolve) => {
```

Comment:
Maybe pull the timeout logic into a separate function.  Something like

```
function (timeout) { return (value) => new Promise((resove) => setTimeout(() => resolve(value), timeout)); }
```

And then you can do like `somePromise().then(wait(10)).then(...`